t/t7519-status-show-ignored-directory.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+#
+#
+
+test_description='git status collapse ignored'
+
+. ./test-lib.sh
+
+
+cat >.gitignore <<\EOF
+*.ign
+ignored_dir/
+!*.unignore
+EOF
+
+test_expect_success 'setup' '
+	mkdir -p tracked tracked_ignored \
+		ignored untracked \
+		untracked_ignored \
+		ignored_dir \
+		dir/ignored_dir \
+		dir/dir/tracked_ignored dir/dir/ignored \
+		dir/dir/untracked dir/dir/untracked_ignored \
+		dir/dir/ignored_dir \
+		dir/dir/ignored_with_sub_untracked \
+		dir/dir/ignored_with_sub_untracked/untracked \
+		dir/dir/untracked_with_sub_ignored \
+		dir/dir/untracked_with_sub_ignored/ignored &&
+
+	touch tracked/tracked \
+		ignored/ignored_1.ign \
+		ignored/ignored_2.ign tracked_ignored/tracked_1 \
```

Comment:
I have a really hard time wrapping my head around these test cases. Judging from the name, `tracked_1` should be tracked, but as `ignored/` is ignored, it is not (and the test below seems to verify that).

Maybe we can come up with a concise list of cases we actually want to test, such as:

- tracked,
- ignored,
- untracked,
- in an ignored subdirectory:
  - tracked,
  - ignored,
  - untracked
- anything more?